schema-derivatives-options.graphql

```diff
@@ -0,0 +1,1252 @@
+  DYNAMIC_LP_FEE
+
+  " Fixed fee that's paid to the protocol, as a percentage of the traded amount. e.g. 0.05% for Sushiswap, 0.02% for Curve v1. "
+  FIXED_PROTOCOL_FEE
+
+  " Some protocols use dynamic protocol fees. Set `feePercentage` as 0 but handle the dynamic fees in the mapping code. "
+  DYNAMIC_PROTOCOL_FEE
+
+  " One-time fee charged by the protocol during deposit, in percentages of the deposit token "
+  DEPOSIT_FEE
+
+  " One-time fee charged by the protocol during withdrawal, in percentages of the withdrawal token "
+  WITHDRAWAL_FEE
+}
+
+type LiquidityPoolFee @entity {
+  " { Fee type }-{ Pool address } "
+  id: ID!
+
+  " Fee as a percentage of the trade (swap) amount. Does not always apply "
+  feePercentage: BigDecimal
+
+  " Type of fee this pool uses "
+  feeType: LiquidityPoolFeeType!
+}
+
+#############################
+##### Protocol Metadata #####
+#############################
+
+interface Protocol {
+  " Smart contract address of the protocol's main contract (Factory, Registry, etc) "
+  id: ID!
+
+  " Name of the protocol, including version. e.g. Uniswap v3 "
+  name: String!
+
+  " Slug of protocol, including version. e.g. uniswap-v3 "
+  slug: String!
+
+  " Version of the subgraph schema, in SemVer format (e.g. 1.0.0) "
+  schemaVersion: String!
+
+  " Version of the subgraph implementation, in SemVer format (e.g. 1.0.0) "
+  subgraphVersion: String!
+
+  " Version of the methodology used to compute metrics, loosely based on SemVer format (e.g. 1.0.0) "
+  methodologyVersion: String!
+
+  " The blockchain network this subgraph is indexing on "
+  network: Network!
+
+  " The type of protocol (e.g. DEX, Lending, Yield, etc) "
+  type: ProtocolType!
+
+  ##### Quantitative Data #####
+
+  " Current TVL (Total Value Locked) of the entire protocol "
+  totalValueLockedUSD: BigDecimal!
+
+  " Revenue claimed by suppliers to the protocol. LPs on DEXs (e.g. 0.25% of the swap fee in Sushiswap). Depositors on Lending Protocols. NFT sellers on OpenSea. "
+  cumulativeSupplySideRevenueUSD: BigDecimal!
+
+  " Gross revenue for the protocol (revenue claimed by protocol). Examples: AMM protocol fee (Sushi’s 0.05%). OpenSea 10% sell fee. "
+  cumulativeProtocolSideRevenueUSD: BigDecimal!
+
+  " All revenue generated by the protocol. e.g. 0.30% of swap fee in Sushiswap, all yield generated by Yearn. "
+  cumulativeTotalRevenueUSD: BigDecimal!
+
+   " All premium paid to enter a position. "
+  cumulativeEntryPremiumUSD: BigDecimal!
+
+  " All premium paid to exit a position. "
+  cumulativeExitPremiumUSD: BigDecimal!
+
+  " All premium paid. "
+  cumulativeTotalPremiumUSD: BigDecimal!
+
+  " All premium paid to deposit in liquidity pools. "
+  cumulativeDepositPremiumUSD: BigDecimal!
+
+  " All premium paid to withdraw from liquidity pools. "
+  cumulativeWithdrawPremiumUSD: BigDecimal!
+
+  " All premium paid to enter and exit liquidity pool. "
+  cumulativeTotalLiquidityPremiumUSD: BigDecimal!
```

Comment:
GMX does something like this. they use it to rebalance the pool (incentivize and disincentivize certain assets being removed). In amalgamated options pools this could also be used.